docs/reference/contributing/guidelines/workflow.md

```diff
@@ -79,6 +87,43 @@ Each feature has a tech lead. This person is responsible for:
 - Rebasing often to track master development.
 - Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
 
+##### Pull request types
+
+We consider the following pull request types.
+
+##### Fix
+
+A bug fix is backward-compatible internal change that fixes incorrect behavior. The fix provides a description what and how it fixes the issue. Tests are included to catch the issue in the future and confirms the correct behavior.
+
+Release: patch
+
+##### Refactor
+
+Refactors are intended for feature releases if they are not fixing specific issue as they 
+can introduce new issues.
+
+Release: feature
+
+##### New target
+
+Adding a new target targets the patch release as it updating targets folder implementation.
+
+Release: patch
+
+##### Feature
+
+New features targets feature releases. It can be integrated only if the feature supports most of the targets (if it requires new target HAL implementation).
+
+Adding a new functionality is considered to be a feature. It does not matter if it is C++, C or Python.
+
+Release: feature
+
+##### Breaking change
+
+Any change that results in breaking user space. It should have strong justification to be considered. Often the change could be made backward compatible, for example deprecate the old functionality and introduce the new replacement.
+
+Release: feature
```

Comment:
As I added versioning here, this should be major version to be correct.

Reply:
I don't believe mbed OS 5 has ever attempted to truly follow semantic versioning, as mbed OS 3 components did - there's been some flexibility on this one. My understanding is we don't deliberately totally break stuff, like binning complete APIs, but we do allow incompatible changes at 5.X to correct API flaws if the deprecation alternative isn't straightforward. (Eg errno mapping changes).

My understanding was that there was an expectation that apps may need to make some changes for a feature release, and as such inclusion version references should be explicit, rather than automatically "take latest" like Yotta, but it should be straightforward.

We're a bit Linux kernely with version number in that regard, rather than user-space shared library. The patch boundary is relatively clear, but when to increment the top digit less so. I have a suspicion the top digit is reserved for marketing.

Reply:
> I have a suspicion the top digit is reserved for marketing.

**ಠ_ಠ**

Reply:
That sounds correct. I don't think we do our major releases quick enough to strictly enforce semanic versioning, specifically in regards to deprecation warnings.

Reply:
@kjbracey-arm Agree. We were flexible (we keep backward compatibility as long as we can, but some breaking changes were done and were part of feature releases). 

What goes to major release then?